### PR TITLE
Refactored bookmarks, added local stored layer settings

### PR DESCRIPTION
**Pull request by almostcake**
Replaced bookmarks storage with unified local storage. Refactored spatial bookmarks. Added mixin for binding properties from model to local storage and method in route to init layers from local ones.

### CODE REVIEW

File: addon/initializers/local-storage.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export function initialize(application) {
+  application.inject('component:spatial-bookmark', 'local-storage-service', 'service:local-storage');
+}
+
```
Comment (kuzkok) [inline addon/initializers/local-storage.js]: need doc comment

File: addon/mixins/flexberry-ddau-slider-actions-handler.js [hunk 2]
```diff
@@ -56,6 +58,9 @@ export default Ember.Mixin.create({
         mutablePropertyPathType === 'string');
 
       setRecord(this, mutablePropertyPath, e.newValue);
+
+      // Save changes to local storage.
+      this.mutateStorage('opacity', mutablePropertyPath, e.newValue);
```
Comment (kuzkok) [inline addon/mixins/flexberry-ddau-slider-actions-handler.js]: we can get bag here if app programmer extend class with this mixin and dont extend with local-storage-binding.

File: addon/routes/edit-map.js [hunk 3]
```diff
@@ -118,5 +137,36 @@ export default EditFormRoute.extend({
     }
 
     return result;
+  },
+
+  /**
+    Recursively creates layers hierarchy.
```
Comment (kuzkok) [inline addon/routes/edit-map.js]: are comment correspond to method work? it seems you ctrl+c\ctrl-v from sortLayersByIndex.

File: addon/routes/edit-map.js [hunk 4]
```diff
@@ -118,5 +137,36 @@ export default EditFormRoute.extend({
...
+
+    @method initLayersFromLocalStorage
+    @param mapId {String}
+    @param layers FlexberryGisMapLayer[]
+  */
+  initLayersFromLocalStorage(mapId, layers) {
+    let localLayers = this.get('service').getFromStorage('layers', mapId);
+
+    for (let local of localLayers) {
+      // Find suitable layer.
```
Comment (kuzkok) [inline addon/routes/edit-map.js]: Need you check for hasOwnProperties?

File: addon/utils/extended-set.js [hunk 5]
```diff
@@ -53,6 +53,42 @@ let setRecord = function (source, keyName, value) {
...
+
+  ```
+*/
+let setAndAssign = function (source, keyName, value) {
+  // Array of keys.
+  let keys = keyName.split('.');
+  let keyPath = keys[0];
+
+  // Create empty object if object in path could not be found or was destroyed.
+  for (let i = 1, len = keys.length; i < len; i++) {
```
Comment (kuzkok) [inline addon/utils/extended-set.js]: len is unnecessary here, complicates understanding

File: addon/mixins/flexberry-ddau-slider-actions-handler.js (hunk 2 above)
Comment (almostcake) [reply addon/mixins/flexberry-ddau-slider-actions-handler.js]: So, what do you propose? Extending mixin with mixin is bad idea. This way, edit-map controller uses local-storage-binding, so applied programmer do not have to do it.

**Pull request merged**